### DISCUSSION TITLE
Getting info from TestReport

### DISCUSSION THREAD

**Discussion by lbrodziak** [pytest-dev/pytest#9950] [Q&A]
Hi,

I would like to access information which is presented in the details section of the pytest html report. Can this be achieved and how? I tried to access it through: pytest_html.extras.html but I don't know if this is a right way. All I need is to get the string that is passed to the  div of class "log".

**Comment by portyu9** [comment]
Hi @lbrodziak 

`pytest_html.extras.html()` is for **adding** HTML to a pytest-html report; it does not retrieve the existing contents of pytest-html's `div.log`.

The underlying pytest `TestReport` already exposes the captured information:

```python
def pytest_runtest_logreport(report):
    if report.when == "call":
        print(report.caplog)
        print(report.capstdout)
        print(report.capstderr)
        print(report.sections)
```

The useful properties are:

- `report.caplog`
- `report.capstdout`
- `report.capstderr`
- `report.sections`

If you specifically need to modify the HTML that pytest-html is going to put in its result details section, use pytest-html's `pytest_html_results_table_html(report, data)` hook instead of scraping its generated `<div class="log">`.

Also, with current pytest-html, use `report.extras` when adding extras; the older `report.extra` spelling is deprecated.

Please mark this answer as accepted if it helped, thank you.
